### DISCUSSION TITLE
Analyzing ancestry on females only

### DISCUSSION THREAD

**Discussion by Mduranton** [tskit-dev/pyslim#158] [Q&A] [answered]
Hi,
I am following on the question I asked on the slim group discussion on how to analyze ancestry only on females after simulating admixture between two populations with sexual chromosomes.

As suggested I used the code :
```
female_nodes = []
for ind in ts.individuals():
  if ind.metadata['sex'] == pyslim.INDIVIDUAL_TYPE_FEMALE:
    female_nodes.extend(ind.nodes)

female_ts = ts.simplify(female_nodes) 
```
to remove males from the tree seq file within python.

However we realized that in some simulations ancestry was over one. It looks like since males from the first generation are removed from the tree seq file, some tree that should have coalesced within these individuals instead coalesce in the next generation meaning in the hybrid population. Then when simplified these trees claim to root to population 2,  not population 0 or 1 which would be the ancestral populations.

to go around that we tried to add a line :
```
female_and_first_gen_nodes = list(female_nodes)
for ind in ts.first_generation_individuals():
        female_and_first_gen_nodes.extend(ind.nodes)
```
I just wanted to check if this was the correct solution. Also when I tried to use SLiM 3.6 I had a message saying that the function first_gen_nodes is no longer supported. Is there something else to use to be sure the script will still be working with a new version of SLiM?

Thanks in advance,

Best wishes,

Maud

**Comment by petrelharp** [comment, marked as answer]
Hi, Maud! Let's see. Simplification won't change where coalescence happens; but it will remove everything above the MRCA of the samples. If I understand correctly, the problem is that, for instance, the females have a MRCA who is in the hybrid population, so after simplification you can't trace back their ancestry to population 0 or 1?  If so, the solution is easy: use the ["keep_input_roots=True" option to simplify](https://tskit.readthedocs.io/en/latest/python-api.html#tskit.TreeSequence.simplify), and those initial nodes will be still there after simplification.  So, just
```
female_ts = ts.simplify(female_nodes, keep_input_roots=True) 
```

Let me know if that solves your problem? If not I think I'll need some more information about how you're calling ancestry.

**Reply by Mduranton** [reply]
Hi Peter,
Thank you very much it works perfectly fine!
Best,
Maud
